src/workerd/server/server.c++

```diff
@@ -2423,9 +2424,17 @@ static kj::Maybe<WorkerdApi::Global> createBinding(
     }
     case config::Worker::Binding::MEMORY_CACHE: {
       auto cache = binding.getMemoryCache();
-      KJ_REQUIRE(cache.hasId() && cache.hasLimits());
+      // TODO(cleanup): Should we have some reasonable default for these so they can
+      // be optional?
+      KJ_REQUIRE(cache.hasLimits());
       Global::MemoryCache cacheCopy;
-      cacheCopy.cacheId = kj::str(cache.getId());
+      // The id is optional. If provided, then multiple bindings with the same id will
+      // share the same cache. Otherwise, a unique id is generated for the cache.
+      if (cache.hasId()) {
+        cacheCopy.cacheId = kj::str(cache.getId());
+      } else {
+        cacheCopy.cacheId = randomUUID(kj::none);
```

Comment:
Instead of randomizing the ID in order to guarantee uniqueness, how about we just don't put the object into the map in the first place? We should construct a `SharedMemoryCache` object and give it directly to the specific Worker.

This does of course require that the Worker holds a strong reference to the `SharedMemoryCache`, which goes back to my other comment about making it atomic-refcounted.

Reply:
Yep, was thinking exactly the same :-) ... Only went with this simply because of how the map *currently* works. If the cache is not shared, there's no reason for an id at all.